src/borg/archiver.py

```diff
@@ -1162,7 +1162,7 @@ def print_text_output(diff, path):
 
         matcher = self.build_matcher(args.patterns, args.paths)
 
-        diffs = Archive.compare_archives_iter(archive1, archive2, matcher, can_compare_chunk_ids=can_compare_chunk_ids, content_only=args.content_only)
+        diffs = Archive.compare_archives_iter(self.print_warning, archive1, archive2, matcher, can_compare_chunk_ids=can_compare_chunk_ids, content_only=args.content_only)
```

Comment:
This is the only caller according to `grep`. By compatible, you mean make it the _last_ required kwarg?

I can't inject the warning printer into `Archive` because `compare_archives_iter` is a `staticmethod`. And I can't return errors to be printed by the caller because `compare_archives_iter` is a generator that returns paths and diffs.

Reply:
I usually make such stuff a kwarg (and have it at the end, after positional args). If one wants to make sure it is always given using the keyword, one can also put `*` into the args list:

```
def compare_archives_iter(archive1, archive2, matcher, *, can_compare_chunk_ids=..., content_only=..., print_warning=...)
```